htlcswitch/switch_test.go

```diff
@@ -2954,6 +2967,7 @@ func checkHtlcEvents(t *testing.T, events <-chan interface{},
 	for _, expected := range expectedEvents {
 		select {
 		case event := <-events:
+			fixExpectedSettleEvent(expected, event)
```

Comment:
The approach you've got here is about as good as we can get in terms of testing, nice job! Only change I'd make is to rename `fixExpectedSettleEvent` -> `setExpectedPreimage` so it's a bit more informative.

Reply:
chnaged